src/pages/settings/PasswordPage.js

```diff
@@ -51,54 +50,109 @@ class PasswordPage extends Component {
             currentPassword: '',
             newPassword: '',
             confirmNewPassword: '',
-            isPasswordRequirementsVisible: false,
-            shouldShowPasswordConfirmError: false,
+            errors: {
+                currentPassword: false,
+                newPassword: false,
+                confirmNewPassword: false,
+                confirmPasswordMatch: false,
+                newPasswordSameAsOld: false,
+            },
         };
 
-        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.submit = this.submit.bind(this);
+        this.getErrorText = this.getErrorText.bind(this);
+        this.validate = this.validate.bind(this);
+        this.clearErrorAndSetValue = this.clearErrorAndSetValue.bind(this);
         this.currentPasswordInputRef = null;
+
+        this.errorKeysMap = {
+            currentPassword: 'passwordPage.errors.currentPassword',
+            confirmNewPassword: 'passwordPage.errors.confirmNewPassword',
+            newPasswordSameAsOld: 'passwordPage.errors.newPasswordSameAsOld',
+            confirmPasswordMatch: 'setPasswordPage.passwordsDontMatch',
+            newPassword: 'passwordPage.errors.newPassword',
+        };
     }
 
     componentWillUnmount() {
         clearAccountMessages();
     }
 
-    onBlurNewPassword() {
-        const stateToUpdate = {};
-        if (!this.state.newPassword || !this.isValidPassword()) {
-            stateToUpdate.isPasswordRequirementsVisible = true;
-        } else {
-            stateToUpdate.isPasswordRequirementsVisible = false;
+    /**
+     * Return the error message for the field
+     *
+     * @param {String} field
+     * @returns {String}
+     */
+    getErrorText(field) {
+        if (this.state.errors[field]) {
+            return this.props.translate(this.errorKeysMap[field]);
         }
+        return null;
```

Comment:
If the doc says we are returning a string then don't return `null` return an empty string.